static/app/utils/reactQuery.tsx

```diff
@@ -0,0 +1,70 @@
+import * as reactQuery from '@tanstack/react-query';
+import {QueryClientConfig} from '@tanstack/react-query';
+
+import RequestError from 'sentry/utils/requestError/requestError';
+import useApi from 'sentry/utils/useApi';
+
+type QueryKeyEndpointOptions = {
+  query?: Record<string, any>;
+};
+
+type QueryKey = readonly [string] | readonly [string, QueryKeyEndpointOptions];
```

Comment:
Can we give these parameters names?
```suggestion
type QueryKey = 
  | readonly [url: string]
  | readonly [url: string, options: QueryKeyEndpointOptions];
```

Reply:
+1 on that